k8sutils/statefulset.go

```diff
@@ -486,14 +519,28 @@ func getEnvironmentVariables(role string, enabledMetric bool, enabledPassword *b
 		}
 	}
 
-	envVars = append(envVars, corev1.EnvVar{
-		Name:  "REDIS_ADDR",
-		Value: redisHost,
-	})
+	RedisPassword := "REDISCLI_AUTH"
+	if enabledMetric {
+		envVars = append(envVars, corev1.EnvVar{
+			Name: "REDIS_ADDR",
+			Value: func(role string, tls *redisv1beta1.TLSConfig) string {
+				prefix := "redis://localhost:"
+				if tls != nil {
+					prefix = "rediss://localhost:"
+				}
+				if role == "sentinel" {
+					return prefix + strconv.Itoa(sentinelPort)
+				} else {
+					return prefix + strconv.Itoa(redisPort)
+				}
+			}(role, tlsConfig),
+		})
+		RedisPassword = "REDIS_PASSWORD"
+	}
 
 	if enabledPassword != nil && *enabledPassword {
 		envVars = append(envVars, corev1.EnvVar{
-			Name: "REDIS_PASSWORD",
+			Name: RedisPassword,
```

Comment:
What's the need of updating this?

Reply:
redis-cli uses [REDISCLI_AUTH](https://github.com/redis/redis/blob/12826fa38f0c481cce36f30b96b82028b1def0cb/src/redis-cli.c) by default
`NOTE: For security reasons, provide the password to redis-cli automatically via the REDISCLI_AUTH environment variable.`
redis_exporter uses [REDIS_PASSWORD](https://github.com/oliver006/redis_exporter/blob/48b92fc8d37f83b70ded121fc78f833e904ea5c4/main.go#L59) by default.